### PR TITLE
JULIA_DEPOT_PATH: expand empty path to default

### PR DESCRIPTION
**Pull request by StefanKarpinski**
This makes the behavior for JULIA_DEPOT_PATH expansion match that of
JULIA_LOAD_PATH expansion and allows one to manipulate the path 
uncondionally with shell expressions like this:
```sh
export JULIA_DEPOT_PATH="$JULIA_DEPOT_PATH:/system/depot"
```
It also changes JULIA_LOAD_PATH expansion to do user name expansion just
like JULIA_DEPOT_PATH does, bringing the two variables into closer
behavioral agreement.

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
On top of https://github.com/JuliaLang/julia/pull/31010

File: base/initdefs.jl [hunk 1]
```diff
@@ -41,16 +41,63 @@ isinteractive() = (is_interactive::Bool)
...
+loading mechanisms, look for package registries, installed packages, named
+environments, repo clones, cached compiled package images, and configuration
+files. By default it includes:
+
+1. `~/.julia` where `~` is the user home as appropriate on the system;
+2. an architecture-specific shared system directory, e.g. `/usr/local/share/julia`;
+3. an architecture-independent shared system directory, e.g. `/usr/share/julia`.
+
+So `DEPOT_PATH` might be:
+```
```
Comment (fredrikekre) [inline base/initdefs.jl]:
```suggestion
```julia
```

**Review by fredrikekre — APPROVED** [review]

File: base/initdefs.jl (hunk 1 above)
Comment (StefanKarpinski) [reply base/initdefs.jl]: Don't we assume Julia code by default in our own docstrings?

Comment (fredrikekre) [reply base/initdefs.jl]: No.

Comment (StefanKarpinski) [reply base/initdefs.jl]: Why not?

**Comment by JeffBezanson** [comment]
Triage approves.

**Comment by fredrikekre** [comment]
Pushed some test cases.

File: base/initdefs.jl (hunk 1 above)
Comment (mortenpi) [reply base/initdefs.jl]: If you don't specify the the language, we assume it's plain text. We _could_ default to `julia`, but then we'd need to introduce `text` or something to remove the highlighting when you don't want it and that feels a bit of a hack.

I also think it is good practice to be explicit about the language -- it makes it easy to copy paste the source between different Markdown processors, the `.md` files on GitHub render correctly etc.

**Comment by StefanKarpinski** [comment]
Thank you, @fredrikekre! You are the best.

**Pull request merged**